internal/eventlog/cursor/cursor.go

```diff
@@ -0,0 +1,100 @@
+// Package cursor implements time-ordered item cursors for an event log.
+package cursor
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// A Source produces cursors based on a time index generator and a sequence
+// counter. A zero-valued Source is ready for use with defaults as described.
+type Source struct {
+	// This function is called to produce the current time index.
+	// If nil, it defaults to time.Now().UnixNano().
+	TimeIndex func() int64
```

Comment:
I sort of feel like I want these to be actual time.Times, though I anticipate being readily convinced of why they shouldn't, it's not obvious to me from just reading the code.

Reply:
They certainly could—the main reason I used integers is to avoid having to hit the struct conversion on the the most-common operations on them in this library (subtracting and converting to string). It also makes it a little easier to write test cases. That said, if there's a reason it matters, it's not hard to change.